sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1136,6 +1155,63 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates the executable graph node inputs & outputs from a topologically
+identical modifiable graph. A topologically identical graph is one with the
+same structure of nodes and edges, and the nodes added in the same order to
+both graphs. Equivalent nodes in topologically identical graphs each have the
+same command, targeting the same device. There is the additional limitation that
+to update an executable graph, every node in the graph must be either a kernel
+command or a host task.
```

Comment:
Question for @Bensuo - how do you feel about removing the "host task" functionality from this to reduce the scope of the initial prototype so it's only kernel commands? (even if host-tasks would be technically implementable now).

Earlier on in `==== Executable Graph Update [[executable-graph-upate]]` description we say

> The only type of nodes that are currently supported for updating in a graph are
kernel execution nodes.

So if not we'll need to update that wording in this PR (not that it's a big deal)

Reply:
Yeah I think it makes sense to remove that for now and implement it later.

Reply:
I removed the host task reference. There was another line a bit further down that I also removed. It said:

> Differences in the following characteristics between two graphs during an
update results in undefined behavior:
**Modifying the native C++ callable of a `host task` node.**